WEBVTT
Kind: captions
Language: ar

00:00:00.000 --> 00:00:02.500
كما رأيتم في الفيديو السابق،

00:00:02.600 --> 00:00:05.633
إن محاولة معالجة أي سلسلة أحداث قد تحدث

00:00:05.733 --> 00:00:09.233
عند التعامل مع مُدخل غير متوقّع
قد تكون صعبة قليلاً.

00:00:09.333 --> 00:00:12.967
هناك في الواقع طريقة أبسط بكثير
للتعامل مع هذا في لغة "Python".

00:00:13.067 --> 00:00:18.467
ستتعلمون هنا كيفية معالجة الأخطاء
باستخدام كتل "try" و"except".

00:00:18.567 --> 00:00:22.533
أولاً، لنتعلّم المزيد
عن ماهية الأخطاء في "Python".

00:00:22.633 --> 00:00:26.300
رأينا رسائل أخطاء لأسباب عديدة في هذه الدورة،

00:00:26.400 --> 00:00:29.067
ويمكننا تقسيمها إلى نوعين:

00:00:29.167 --> 00:00:31.933
أخطاء نص برمجيّ واستثناءات.

00:00:32.033 --> 00:00:35.300
تحصل أخطاء النص البرمجيّ
عندما لا تستطيع "Python" ترجمة الكود

00:00:35.400 --> 00:00:39.700
لأننا لم نتّبع
النص البرمجيّ الصحيح للغة "Python".

00:00:39.800 --> 00:00:43.333
هذه أخطاء ستواجهونها عادة
عندما تقترفون خطأ إملائياً،

00:00:43.433 --> 00:00:45.433
أو عندما تبدؤون في تعلم "Python".

00:00:45.533 --> 00:00:51.133
ظهر لي هنا خطأ نص برمجيّ لأنني نسيت
علامة التنصيص في نهاية السلسلة النصيّة.

00:00:51.233 --> 00:00:53.167
يمكنني رؤية هذا في رسالة الخطأ:

00:00:53.267 --> 00:00:57.533
SyntaxError: End of line
when scanning string literal.

00:00:57.633 --> 00:01:00.033
أما النوع الآخر من الأخطاء، وهو الاستثناءات،

00:01:00.133 --> 00:01:04.400
يظهر عند حدوث
أشياء غير متوقّعة خلال تنفيذ الكود.

00:01:04.833 --> 00:01:09.533
هنا، بالرغم من صحة النص البرمجيّ للكود،
إلا أننا نواجه استثناءً

00:01:09.633 --> 00:01:11.967
لأن السلسلة نصية التي تحمل الكلمة "ten"

00:01:12.067 --> 00:01:14.800
ليست وسيطة صالحة لدالة "int".

00:01:14.900 --> 00:01:17.933
هناك أنواع مختلفة
من الاستثناءات المضمنة في لغة "Python"،

00:01:18.033 --> 00:01:22.667
ويمكنكم أن تروا
أي استثناءات تم عرضها هنا هي رسالة الخطأ.

00:01:22.767 --> 00:01:25.567
خطأ القيمة، "ValueError"،
هو أحد أنواع الاستثناءات.

00:01:25.667 --> 00:01:29.767
ويحدث عندما يتم تزويد
عملية مضمنة أو دالة مضمنة

00:01:29.867 --> 00:01:34.400
بوسيطة من النوع الصحيح لكن بقيمة غير مناسبة.

00:01:34.500 --> 00:01:37.167
هذا مثال آخر على وجود استثناء.

00:01:37.267 --> 00:01:41.367
نحاول هنا الإشارة
إلى اسم متغيّر لم نقم بتعريفه بعد.

00:01:41.467 --> 00:01:43.967
فنحصل على استثناء من نوع "NameError".

00:01:44.067 --> 00:01:47.267
مجدداً، هذا الكود صحيح من حيث النص البرمجيّ،

00:01:47.367 --> 00:01:50.000
لكن عندما تصل "Python"
إلى هذا السطر من الكود،

00:01:50.100 --> 00:01:53.433
لا تستطيع إيجاد القيمة لهذا المتغيّر
عندما تحاول تنفيذه.

00:01:53.533 --> 00:01:56.467
؟؟
فتظهر استثناءً وتوقف تشغيل الكود.

